backend/api/v1/role_service.go

```diff
@@ -109,6 +109,13 @@ func (s *RoleService) DeleteRole(ctx context.Context, request *v1pb.DeleteRoleRe
 	if role == nil {
 		return nil, status.Errorf(codes.NotFound, "role not found: %s", roleID)
 	}
+	using, project, err := s.store.GetProjectUsingRole(ctx, api.Role(roleID))
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to check if the role is used: %v", err)
+	}
+	if using {
```

Comment:
What if a principal is archived, however, its project member cannot be deleted from the project?

Reply:
I think you can delete the binding of an archived principal and some roles.